frontend/src/hooks/useFetchNextPage.tsx

```diff
@@ -0,0 +1,28 @@
+import { useEffect } from 'react';
+import { useInView } from 'react-intersection-observer';
+
+interface IProps {
+  hasNextPage?: boolean;
+  fetchNextPage: () => void;
+}
+
+const useFetchNextPage: (props: IProps) => (node?: Element | null | undefined) => void = ({
```

Comment:
Seems we need to avoid duplicating of the `undefined`, it's already presents with ? symbol:
```suggestion
const useFetchNextPage: (props: IProps) => (node?: Element | null) => void = ({
```